lwgenerate/gcp/gcp.go

```diff
@@ -0,0 +1,533 @@
+package gcp
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type ExistingServiceAccountDetails struct {
+	// Existing Service Account Name
+	Name string
+
+	// Existing Service Account private key in JSON format, base64 encoded
+	PrivateKey string
+}
+
+// NewExistingServiceAccountDetails Create new existing Service Account details
+func NewExistingServiceAccountDetails(name string, privateKey string) *ExistingServiceAccountDetails {
+	return &ExistingServiceAccountDetails{
+		Name:       name,
+		PrivateKey: privateKey,
+	}
+}
+
+type GenerateGcpTfConfigurationArgs struct {
+	// Should we configure AuditLog integration in LW?
+	AuditLog bool
+
+	// Should we configure CSPM integration in LW?
+	Config bool
+
+	// Path to service account credentials to be used by Terraform
+	ServiceAccountCredentials string
+
+	// Should we configure an Organization wide integration?
+	OrganizationIntegration bool
+
+	// Supply a GCP Organization ID, only asked if OrganizationIntegration is True
+	GcpOrganizationId string
```

Comment:
nit: Do we need both of these?  Could the presence of an org id set org integration to true? I noticed you have to supply both when using an org and in most cases, we are testing the presence of both of these values so it seems it may be redundant.

Reply:
Fair question... I was simply copying the convention we have within the TF modules. 

I guess we can infer the value that we should be setting the Terraform `org_integration` var boolean to, based on whether the user supplies an organization id.
Any objections to this approach @afiune @dmurray-lacework ? I can't think of a scenario where we would be setting the organization ID and it not being an org integration 🤔

Reply:
Going to leave this one as-is for now. I may revisit when it comes to implementing the survey, as it mightn't make sense to have the boolean as part of the survey